File: game.r
# Требуется библиотека grid
library(grid)

# Настройки
grid.newpage()
player_x <- 0.5
player_speed <- 0.02

# Отрисовка мира (упрощенная статическая)
grid.rect(y=0.25, height=0.5, gp=gpar(fill="lightblue", col=NA)) # Небо
grid.rect(y=0.5, height=0.1, gp=gpar(fill="green", col=NA))     # Трава
grid.rect(y=0.3, height=0.3, gp=gpar(fill="brown", col=NA))     # Земля

# Игровой цикл и управление (нажмите ESC в консоли для выхода)
while(TRUE) {
  # В R интерактивное управление обычно привязывается к графическим окнам tcltk
  # Ниже пример обновления позиции игрока
  grid.rect(x=player_x, y=0.58, width=0.05, height=0.05, gp=gpar(fill="red"))
  Sys.sleep(0.05)
  # Для полноценного ввода в R нужно использовать библиотеку 'tcltk'
}

File: game.py
import pygame
import sys
import help  # Импортируем наш новый файл help.py

# Инициализация библиотеки
pygame.init()

# Настройки окна игры
WIDTH, HEIGHT = 800, 600
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption("Мини-Майнкрафт 2D с подсказками")

# Цвета (RGB)
SKY_BLUE = (135, 206, 235)
GREEN = (34, 139, 34)
BROWN = (139, 69, 19)
GRAY = (128, 128, 128)
RED = (255, 0, 0)

# Размеры блоков и сетки
BLOCK_SIZE = 40
COLS = WIDTH // BLOCK_SIZE
ROWS = HEIGHT // BLOCK_SIZE

# Генерация плоского мира
world = []
for row in range(ROWS):
    world_row = []
    for col in range(COLS):
        if row < ROWS // 2:
            world_row.append(0)  
        elif row == ROWS // 2:
            world_row.append(1)  
        elif row < ROWS // 2 + 3:
            world_row.append(2)  
        else:
            world_row.append(3)  
    world.append(world_row)

# Параметры игрока
player_x = WIDTH // 2
player_y = (ROWS // 2 - 1) * BLOCK_SIZE
player_speed = 5

# Игровой цикл
clock = pygame.time.Clock()
running = True

while running:
    clock.tick(60)
    
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
            
    # Управление персонажем
    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT] and player_x > 0:
        player_x -= player_speed
    if keys[pygame.K_RIGHT] and player_x < WIDTH - BLOCK_SIZE:
        player_x += player_speed

    # Очистка экрана и заливка небом
    screen.fill(SKY_BLUE)

    # Отрисовка блоков мира
    for row in range(ROWS):
        for col in range(COLS):
            block_type = world[row][col]
            x = col * BLOCK_SIZE
            y = row * BLOCK_SIZE
            
            if block_type == 1:
                pygame.draw.rect(screen, GREEN, (x, y, BLOCK_SIZE, BLOCK_SIZE))
            elif block_type == 2:
                pygame.draw.rect(screen, BROWN, (x, y, BLOCK_SIZE, BLOCK_SIZE))
            elif block_type == 3:
                pygame.draw.rect(screen, GRAY, (x, y, BLOCK_SIZE, BLOCK_SIZE))

    # Отрисовка игрока
    pygame.draw.rect(screen, RED, (player_x, player_y, BLOCK_SIZE, BLOCK_SIZE))

    # ЕСЛИ НАЖАТА КЛАВИША 'H' — вызываем функцию меню из help.py
    if keys[pygame.K_h]:
        help.draw_help_menu(screen)

    # Обновление экрана
    pygame.display.flip()

pygame.quit()
sys.exit()
